### PR TITLE
Build on Ubuntu 14.04 + use ethereumjs-testrpc v3

### PR DESCRIPTION
**Pull request by veox**
This is an attempt to get CI working again.

The PR branch is a squashed version of `travis-use-ubuntu14.04+ethereumjs-testrpc3`, which was used to "thrash" Travis with build/job requests. (See [build history](https://travis-ci.org/veox/ens/builds).)

First tried updating the build machine to Ubuntu 14.04 - since 12.04, which the current CI builds are being run on, has [reached EOL](http://fridge.ubuntu.com/2017/05/01/ubuntu-12-04-precise-pangolin-end-of-life-reached-on-april-28-2017/).

If this is undesirable, the appropriate commit should be reverted.

Since this didn't help with Travis jobs failing sporadically and unpredictably (3/10 on average), I updated the `ethereumjs-testrpc` dependency to v3.

That [uses `geth`-like gas rules](https://github.com/ethereumjs/testrpc/releases/tag/v3.0.0), so many calls required specifying that.

Note that I didn't change _all_ the calls, but mainly for cases that were failing. I might've missed some, or specified gas where it wasn't really necessary.

Also, it seems that some calls use high gas, e.g. `invalidateName` and `finalizeAuction`, which isn't mentioned in the docs (I think). This was not fixed.

### CODE REVIEW

File: test/simplehashregistrar_test.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ describe('SimpleHashRegistrar', function() {
 
 	beforeEach(function(done) {
 
-		this.timeout(5000);
+		this.timeout(10000);
```
Comment (veox) [inline test/simplehashregistrar_test.js]: Might be superficial?..

File: test/simplehashregistrar_test.js [hunk 2]
```diff
@@ -804,15 +804,16 @@ describe('SimpleHashRegistrar', function() {
 		var sealedBid = null;
 
 		advanceTimeAsync(launchLength)
-			.then((done) => registrar.startAuctionAsync(web3.sha3('name'), {from: accounts[0]}))			.then((done) => registrar.finalizeAuctionAsync(web3.sha3('name'), {from: accounts[0]}))
+			.then((done) => registrar.startAuctionAsync(web3.sha3('name'), {from: accounts[0], gas: 500000}))
+			.then((done) => registrar.finalizeAuctionAsync(web3.sha3('name'), {from: accounts[0]}))
```
Comment (veox) [inline test/simplehashregistrar_test.js]: How the hell did this get here?!.

Comment (veox) [inline test/simplehashregistrar_test.js]: Oh! It's a long line. Scroll right to see on github. :)

File: test/simplehashregistrar_test.js [hunk 3]
```diff
@@ -1025,7 +1025,7 @@ describe('SimpleHashRegistrar', function() {
 			.then((balance) => { invalidator.startingBalance = balance.toFixed(); })
 			// Advance time past soft launch
 			.then((result) => advanceTimeAsync(launchLength))
-			// Start some auctions (100000 gas for each)
+			// Start some auctions (300000 gas for each)
```
Comment (veox) [inline test/simplehashregistrar_test.js]: Isn't that 100K for each, 300K total?..

Comment (Arachnid) [inline test/simplehashregistrar_test.js]: Oh, oops!

**Pull request merged**